pyannote/audio/labeling/tasks/base.py

```diff
@@ -142,6 +140,10 @@ def __init__(self, feature_extraction, protocol, subset='train',
         self.mask_dimension = mask_dimension
         self.mask_logscale = mask_logscale
 
+        self.nb_sequences_per_epoch = self.batches_per_epoch * (self.batch_size + 2)
+        # Iteration counter
+        self.iteration = 0
```

Comment:
This whole `epoch`-dependent feature extraction should be handled by a callback instead.
Please remove it from here.